docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,198 @@ Example for the `__time` dimension:
   "function" : "function(t) { return 'Second ' + Math.floor((t % 60000) / 1000); }"
 }
 ```
+
+### Namespaced extraction function (EXPERIMENTAL)
+A namespaced extraction function is a simple key-value mapping where the key-value mappings are unique to a particular namespace.
+This can be used for re-naming values in a datasource for certain queries. To use a particular namespace, simply set the appropriate namespace for the dimExtractionFn
+```json
+    "dimExtractionFn" : {
+      "type":"namespace",
+      "namespace":"some_namespace",
+      "missingValue":"NA",
+      "uniqueRenames":false
+    }
+```
+
+A property of `missingValue` can be specified at query time to hint how to handle missing values. Setting `missingValue` to `""` or `null` or omitting the property will cause the dimension's original value to be used.
```

Comment:
We have already defined the null behavior for extraction functions, so we should make an effort to support it, so we don't have more things to fix down the road.

From a user perspective, having to use something other than empty string or null requires them to implement two different behaviors for missing values on the client side, one for regular queries and one for queries with lookups, which is not very user-friendly. If we reserve empty string for that case as missingValue, then it becomes confusing since those will be returned as nulls.

Another things to keep in mind is that the lookup itself may map things to null, and I am not sure if we should have a difference between a missing mappings and a mapping that maps to null, but that's something we may want to discuss.

Also, do we allow mapping `null` to something or not?

I think we need to think about all those cases and decide how we want to handle those edge cases.

Given all this, I would be in favor of having a configurable behavior rather than just a string, with two options:
- `defaultValue` to specify what to use in case there is no mapping, which will allow use of null / empty string to map anything not specified to null
- `leaveAsIs` to tell it not do to anything

the default if unspecified should be to map to null.